src/Service/ValueHandler.php

```diff
@@ -71,144 +84,151 @@ private function getValueByPath(NodeInterface $node, $path, bool $failOnError =
   /**
    * Get serialized field value.
    *
-   * @param \Drupal\Core\Field\FieldItemListInterface $field
+   * @param \Drupal\Core\Field\FieldItemListInterface<FieldItemInterface> $field
    *   The field.
    *
-   * @return null|string
+   * @return mixed
    *   The serialized value.
    */
-  private function getSerializedValueByType(FieldItemListInterface $field) {
+  private function getSerializedValueByType(FieldItemListInterface $field): mixed {
     $value = NULL;
 
-    if ($field) {
-      $fieldDefinition = $field->getFieldDefinition();
-
-      switch ($fieldDefinition->getType()) {
-        case 'image':
-          $uri = empty($field->entity) ? NULL : $field->entity->getFileUri();
-          $value = empty($uri) ? NULL : file_create_url($uri);
-          break;
-
-        case 'link':
-          $value = $field->uri;
-          break;
-
-        case 'entity_reference':
-          $id_array = $field->getValue();
-          $target_type = $fieldDefinition->getItemDefinition()->getSetting('target_type');
-          $ids = [];
-          foreach ($id_array as $array) {
-            foreach (array_values($array) as $id) {
-              $ids[] = $id;
-            };
-          }
+    $fieldDefinition = $field->getFieldDefinition();
 
-          if ('node' == $target_type) {
-            $entities = Node::loadMultiple($ids);
+    switch ($fieldDefinition->getType()) {
+      case 'image':
+        $file = File::load($field->target_id);
+        if ($file) {
+          $uri = $file->getFileUri();
+          $value = empty($uri) ? NULL : \Drupal::service('file_url_generator')->generateAbsoluteString($uri);
```

Comment:
Use dependency injection to get (and set) the `file_url_generator` service in the constructor.